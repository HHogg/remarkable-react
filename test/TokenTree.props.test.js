import buildTokenTree from './buildTokenTree';

describe('TokenTree props', () => {
  test('[align]', () => {
    expect(buildTokenTree(`

| Header 1 | Header 2 | Header 3 |
| :-------- | ---------| --------: |

    `)).toMatchSnapshot();
  });

  test('[alt]', () => {
    expect(buildTokenTree(`

![Alt]()

    `)).toMatchSnapshot();
  });

  test('[block]', () => {
    expect(buildTokenTree(`

\`Code\`

    `)).toMatchSnapshot();
  });

  describe('[content]', () => {
    test('without option html', () => {
      expect(buildTokenTree(`

<p>Hello</p>
<iframe src="/something.html" />

      `)).toMatchSnapshot();
    });

    test('with option html', () => {
      expect(buildTokenTree(`

<p>Hello</p>
<iframe src="/something.html" />

      `, { html: true })).toMatchSnapshot();
    });
  });

  test('[hLevel]', () => {
    expect(buildTokenTree(`

# Heading 1

      `)).toMatchSnapshot();
  });

  test('[href]', () => {
    expect(buildTokenTree(`

[](http://google.com)

    `)).toMatchSnapshot();
  });

  test('[level]', () => {
    expect(buildTokenTree(`

Paragraph

      `)).toMatchSnapshot();
  });

  test('[lines]', () => {
    expect(buildTokenTree(`

Paragraph

      `)).toMatchSnapshot();
  });

  describe('[linkTarget]', () => {
    test('without remarkableOption', () => {
      expect(buildTokenTree(`

[Link](http://link.com "Title")

      `)).toMatchSnapshot();
    });

    test('with remarkableOption', () => {
      expect(buildTokenTree(`

[Link](http://link.com "Title")

      `, { linkTarget: '_blank' })).toMatchSnapshot();
    });
  });

  test('[order]', () => {
    expect(buildTokenTree(`

1. List item 1

      `)).toMatchSnapshot();
  });

  test('[params]', () => {
    expect(buildTokenTree(`

\`\`\`js
Code
\`\`\`

      `)).toMatchSnapshot();
  });

  test('[src]', () => {
    expect(buildTokenTree(`

![](http://google.com)

    `)).toMatchSnapshot();
  });

  test('[title]', () => {
    expect(buildTokenTree(`

![](http://google.com "Title")

    `)).toMatchSnapshot();
  });

  test('[type]', () => {
    expect(buildTokenTree(`

# Heading 1

      `)).toMatchSnapshot();
  });
});

describe('types', () => {
  describe('Function', () => {
    test('String', () => {
      expect(buildTokenTree(`

![](http://google.com "Title")

      `, {}, {
        remarkableProps: {
          title: (value) => ({
            key: 'newTitle',
            value: value.toUpperCase(),
          }),
        },
      })).toMatchSnapshot();
    });

    test('Boolean (true)', () => {
      expect(buildTokenTree(`

![](http://google.com "true")

      `, {}, {
        remarkableProps: {
          title: (value) => ({
            value: value && value.toLowerCase() === 'true',
          }),
        },
      })).toMatchSnapshot();
    });

    test('Boolean (false)', () => {
      expect(buildTokenTree(`

![](http://google.com "false")

      `, {}, {
        remarkableProps: {
          title: (value) => ({
            value: value && value.toLowerCase() === 'true',
          }),
        },
      })).toMatchSnapshot();
    });
  });

  test('String', () => {
    expect(buildTokenTree(`

![](http://google.com "Title")

        `, {}, {
          remarkableProps: {
            title: 'newTitle',
          },
        })).toMatchSnapshot();
  });

  describe('Boolean', () => {
    test('false', () => {
      expect(buildTokenTree(`

![](http://google.com "Title")

      `, {}, {
        remarkableProps: {
          title: false,
        },
      })).toMatchSnapshot();
    });

    test('true', () => {
      expect(buildTokenTree(`

![](http://google.com "Title")

      `, {}, {
        remarkableProps: {
          type: true,
        },
      })).toMatchSnapshot();
    });
  });
});
