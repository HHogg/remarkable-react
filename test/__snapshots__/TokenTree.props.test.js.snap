// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TokenTree props [align] 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "children": Array [
          Object {
            "children": Array [
              Object {
                "children": Array [
                  "Header 1",
                ],
                "props": Object {
                  "style": Object {
                    "textAlign": "left",
                  },
                },
                "type": "th",
              },
              Object {
                "children": Array [
                  "Header 2",
                ],
                "props": Object {},
                "type": "th",
              },
              Object {
                "children": Array [
                  "Header 3",
                ],
                "props": Object {
                  "style": Object {
                    "textAlign": "right",
                  },
                },
                "type": "th",
              },
            ],
            "props": Object {},
            "type": "tr",
          },
        ],
        "props": Object {},
        "type": "thead",
      },
      Object {
        "children": Array [],
        "props": Object {},
        "type": "tbody",
      },
    ],
    "props": Object {},
    "type": "table",
  },
]
`;

exports[`TokenTree props [alt] 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "children": undefined,
        "props": Object {
          "alt": "Alt",
          "src": "",
          "title": "",
        },
        "type": "img",
      },
    ],
    "props": Object {},
    "type": "p",
  },
]
`;

exports[`TokenTree props [block] 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "children": "Code",
        "props": Object {},
        "type": "code",
      },
    ],
    "props": Object {},
    "type": "p",
  },
]
`;

exports[`TokenTree props [content] with option html 1`] = `
Array [
  Object {
    "children": null,
    "props": Object {
      "dangerouslySetInnerHTML": Object {
        "__html": "<p>Hello</p>
<iframe src=\\"/something.html\\" />
",
      },
    },
    "type": "html",
  },
]
`;

exports[`TokenTree props [content] without option html 1`] = `
Array [
  Object {
    "children": Array [
      "<p>Hello</p>",
      undefined,
      "<iframe src=\\"/something.html\\" />",
    ],
    "props": Object {},
    "type": "p",
  },
]
`;

exports[`TokenTree props [hLevel] 1`] = `
Array [
  Object {
    "children": Array [
      "Heading 1",
    ],
    "props": Object {},
    "type": "h1",
  },
]
`;

exports[`TokenTree props [href] 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "children": Array [],
        "props": Object {
          "href": "http://google.com",
          "title": "",
        },
        "type": "a",
      },
    ],
    "props": Object {},
    "type": "p",
  },
]
`;

exports[`TokenTree props [level] 1`] = `
Array [
  Object {
    "children": Array [
      "Paragraph",
    ],
    "props": Object {},
    "type": "p",
  },
]
`;

exports[`TokenTree props [lines] 1`] = `
Array [
  Object {
    "children": Array [
      "Paragraph",
    ],
    "props": Object {},
    "type": "p",
  },
]
`;

exports[`TokenTree props [linkTarget] with remarkableOption 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "children": Array [
          "Link",
        ],
        "props": Object {
          "href": "http://link.com",
          "target": "_blank",
          "title": "Title",
        },
        "type": "a",
      },
    ],
    "props": Object {},
    "type": "p",
  },
]
`;

exports[`TokenTree props [linkTarget] without remarkableOption 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "children": Array [
          "Link",
        ],
        "props": Object {
          "href": "http://link.com",
          "title": "Title",
        },
        "type": "a",
      },
    ],
    "props": Object {},
    "type": "p",
  },
]
`;

exports[`TokenTree props [order] 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "children": Array [
          Object {
            "children": Array [
              "List item 1",
            ],
            "props": Object {},
            "type": "p",
          },
        ],
        "props": Object {},
        "type": "li",
      },
    ],
    "props": Object {},
    "type": "ol",
  },
]
`;

exports[`TokenTree props [params] 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "children": "Code
",
        "props": Object {},
        "type": "code",
      },
    ],
    "props": Object {},
    "type": "pre",
  },
]
`;

exports[`TokenTree props [src] 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "children": undefined,
        "props": Object {
          "alt": "",
          "src": "http://google.com",
          "title": "",
        },
        "type": "img",
      },
    ],
    "props": Object {},
    "type": "p",
  },
]
`;

exports[`TokenTree props [title] 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "children": undefined,
        "props": Object {
          "alt": "",
          "src": "http://google.com",
          "title": "Title",
        },
        "type": "img",
      },
    ],
    "props": Object {},
    "type": "p",
  },
]
`;

exports[`TokenTree props [type] 1`] = `
Array [
  Object {
    "children": Array [
      "Heading 1",
    ],
    "props": Object {},
    "type": "h1",
  },
]
`;

exports[`types Boolean false 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "children": undefined,
        "props": Object {
          "alt": "",
          "src": "http://google.com",
        },
        "type": "img",
      },
    ],
    "props": Object {},
    "type": "p",
  },
]
`;

exports[`types Boolean true 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "children": undefined,
        "props": Object {
          "alt": "",
          "src": "http://google.com",
          "title": "Title",
          "type": "image",
        },
        "type": "img",
      },
    ],
    "props": Object {
      "type": "paragraph_open",
    },
    "type": "p",
  },
]
`;

exports[`types Function Boolean (false) 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "children": undefined,
        "props": Object {
          "alt": "",
          "src": "http://google.com",
          "title": false,
        },
        "type": "img",
      },
    ],
    "props": Object {},
    "type": "p",
  },
]
`;

exports[`types Function Boolean (true) 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "children": undefined,
        "props": Object {
          "alt": "",
          "src": "http://google.com",
          "title": true,
        },
        "type": "img",
      },
    ],
    "props": Object {},
    "type": "p",
  },
]
`;

exports[`types Function String 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "children": undefined,
        "props": Object {
          "alt": "",
          "newTitle": "TITLE",
          "src": "http://google.com",
        },
        "type": "img",
      },
    ],
    "props": Object {},
    "type": "p",
  },
]
`;

exports[`types String 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "children": undefined,
        "props": Object {
          "alt": "",
          "newTitle": "Title",
          "src": "http://google.com",
        },
        "type": "img",
      },
    ],
    "props": Object {},
    "type": "p",
  },
]
`;
